client: add toxics and promote from wip

Sirupsen opened the pull request:
@eapache @xthexder

xthexder commented on client/README.md:
```diff
@@ -0,0 +1,8 @@
+section in the Toxiproxy README](https://github.com/shopify/toxiproxy#usage)
+before attempting to use the client.
+
+For Usage please see the Godoc documentation for the package.
```
It woud be good to actually provide a link to the godocs.

xthexder commented on client/client.go:
```diff
@@ -46,6 +56,8 @@ func (client *Client) Proxies() (map[string]*Proxy, error) {
 }
 
+// NewProxy instantiates a new proxy instance. Note Create() must be called on
+// it to create it.
```
This should probably note that the proxy is not enabled by default

xthexder commented:
Changes look good :+1: 

I noticed while reading through the main Readme again that the Curl example is out of date (it doesn't contain any of the proxy enabled/disabled stuff)
It would probably be a good idea to revisit that at some point.

Sirupsen commented:
Rebased and fixed your comments. Fixed the main README change in  https://github.com/Shopify/toxiproxy/commit/b979c77411aa6336d716a1f074985248a9e56cb2

eapache commented on client/client.go:
```diff
@@ -1,3 +1,5 @@
+// The Toxiproxy Go client provides a wrapper around the Toxiproxy HTTP API for
```
By convention, package-level godoc starts with "Package $NAME"

eapache commented on client/client.go:
```diff
@@ -151,6 +171,29 @@ func (proxy *Proxy) Toxics(direction string) (map[string]interface{}, error) {
 
+// SetToxic sets the parameters for a toxic with a given name in the direction.
+// See https://github.com/Shopify/toxiproxy#toxics for a list of all Toxics.
+func (proxy *Proxy) SetToxic(name string, direction string, fields Fields) (map[string]interface{}, error) {
```
should direction (and maybe name) be enum types rather than arbitrary strings?

eapache commented:
A few minor comments, but :+1:
